### PR TITLE
Put mod_roster config options in a map with defaults

### PR DESCRIPTION
**Pull request by gustawlippa**
Normal stuff + some clean up in comments and specs.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111017) / small_tests / 0cab5ec50b82a1142bbedd0e6dfbc1b74e0e0efc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111017/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111017/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-03_10.23.22/lib.mongooseim.logs/run.2022-03-03_10.23.23/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111198) / small_tests / bb2408a43510b2837d5d8b09108cbc8841fafbc3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111198/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111198/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-03_11.50.54/lib.mongooseim.logs/run.2022-03-03_11.50.55/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111224) / small_tests / c8b2dadb717f2efc3f772700099951de7143a947
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111224/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111224/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-03_12.22.28/lib.mongooseim.logs/run.2022-03-03_12.22.28/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111249) / small_tests / 389d4af53041fff3cfc440a26f836dd161774873
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111249/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111249/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-03_12.50.15/lib.mongooseim.logs/run.2022-03-03_12.50.15/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/111318) / small_tests / fcc214b12d41a1e3be849c134720aba50fbb8ba1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111318/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111318/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-03-03_14.04.53/lib.mongooseim.logs/run.2022-03-03_14.04.54/suite.log.html)

File: test/config_parser_SUITE.erl [hunk 1]
```diff
@@ -2807,27 +2807,27 @@ registration_watchers(JidBins) ->
...
-    ?cfgh(M([{bucket_type, <<"rosters">>}]),
-          T(#{<<"riak">> => #{<<"bucket_type">> => <<"rosters">>}})),
-    ?cfgh(M([{version_bucket_type, <<"roster_versions">>}]),
-          T(#{<<"riak">> => #{<<"version_bucket_type">> => <<"roster_versions">>}})),
+    ?cfgh(P ++ [versioning],  true,
+          T(#{<<"versioning">> => true})),
+    ?cfgh(P ++ [store_current_id], true,
+          T(#{<<"store_current_id">> => true})),
+    ?cfgh(P ++ [backend], rdbms,
+          T(#{<<"backend">> => <<"rdbms">>})),
```
Comment (chrzaszcz) [inline test/config_parser_SUITE.erl:2819]: Could you check that Riak defaults are present if `backend` is set to `riak`?

**Review by chrzaszcz — COMMENTED** [review]
Looks good, nice to see improvements in various modules, I added one comment.

File: test/config_parser_SUITE.erl (hunk 1 above)
Comment (gustawlippa) [reply test/config_parser_SUITE.erl:2819]: Please take a look if the test added is OK

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/111515) / small_tests / 429ebacc86fb9f46eb824b5c00cfb963cbddfc87
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3569/111515/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3569/111515/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-03-04_11.29.30/lib.mongooseim.logs/run.2022-03-04_11.29.31/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**